04-artifact_correction_ica.py

```diff
@@ -12,44 +12,35 @@
 from mne.preprocessing import ICA
 from mne.parallel import parallel_func
 
-from library.config import meg_dir, random_state, N_JOBS
+import config
 
 # Here we always process with the 1 Hz highpass data (instead of using
 # l_freq) because ICA needs a highpass.
 
 
-def run_ica(subject_id, tsss=None):
-    subject = "sub%03d" % subject_id
+def run_ica(subject, tsss=None):
     print("Processing subject: %s%s"
           % (subject, (' (tSSS=%d)' % tsss) if tsss else ''))
-    data_path = op.join(meg_dir, subject)
-    raws = list()
+    meg_subject_dir = op.join(config.meg_dir, subject)
+    raw_fnames = op.join(meg_subject_dir, '%s_audvis_filt_sss_raw.fif' % subject)
     print("  Loading runs")
-    
-    # XXX get the number of runs from the data
-    for run in range(1, 7): 
-        if tsss:
-            run_fname = op.join(data_path, 'run_%02d_filt_tsss_%d_raw.fif'
-                                % (run, tsss))
-        else:
-            run_fname = op.join(data_path, 'run_%02d_filt_sss_highpass-%sHz'
-                                '_raw.fif' % (run, 1))
-        raws.append(mne.io.read_raw_fif(run_fname))
+    raws = [mne.io.read_raw_fif(raw_fnames)]
+
     raw = mne.concatenate_raws(raws)
     # SSS reduces the data rank and the noise levels, so let's include
     # components based on a higher proportion of variance explained (0.999)
     # than we would otherwise do for non-Maxwell-filtered raw data (0.98)
-    n_components = 0.999
+    n_components = 0.999  # XXX: This can bring troubles to ICA
```

Comment:
The right way to set the n_components is this:
```py
meg_maxfilter_rank = raw.copy().pick_types(meg=True).estimate_rank()
ica = mne.preprocessing.ICA(n_components=meg_maxfilter_rank,
                            method=method,
                            random_state=42)
```

Plus there was the problem of centring the PCA. (check https://github.com/massich/epilepsy-biomag-2018)